lib/OpenQA/Parser/Format/JUnit.pm

```diff
@@ -94,11 +99,17 @@ sub parse {
                 $tc_result =~ s/^skipped$/missing/;
                 $tc_result =~ s/^error$/unknown/;     # error in the testsuite itself
                 $tc_result =~ s/^failure$/fail/;      # test failed
+                $tc_result =~ s/^softfail$/softfail/; # test softfailed
```

Comment:
Replacing softfail with softfail is useless. I don't know to which "set of states" the right side corresponds to. The right sides look like test module results. But shouldn't it be fail**ed** and softfail**ed** then (to match the CSS classes defined in `assets/stylesheets/test-details.scss` and substitutions in `OpenQA::Schema::Result::JobModules:::update_result`)?

Reply:
The effective state is "softfail" based on my experiments, otherwise the junit log won't display correct visual effect. As far as this  "$tc_result =~ s/^softfail$/softfail is concerned, it was added for sake of completeness purpose otherwise $tc_result will not get assigned "tc result" which can have some effect on the overall junit result. Actually, you can consider it as safe net if someone does not derive overall result at the testsuite level, he or she can still have chance to obtain correct overall module result from at the testcase level.

Reply:
Ok, so this has nothing to do with openQA's UI and its CSS classes then. In this case I'm fine with the suggestion.

Reply:
Done.

Reply:
Already verification run with this change
10.67.133.40/tests/137